### ISSUE TITLE
Refactor RunContext creation

### ISSUE THREAD

**Issue by loicmathieu** [kestra-io/kestra#5281]
### Issue description

Today, a RunContext is created in the Executor (or the Scheduler), then serialized in a WorkerJob to be re-hydrated in the Worker which in turn serialized it in a WorkerJobResult to be re-hydrated in the Executor.

This dance compexifies a lot how a RunContext is created, and open the door to security issues as the RunContext is a complex object which should not be serialized. In fact, looking at the RunCOntext the only parts that are really needed to be serialized is the variables as they might not be available in the Worker (for ex flow variables).

What we can, instead of serializing the RunContext on itself, serialize the variables (or maybe some kind of light ExecutionContext which contains the variable (and maybe one or two more properties that are really needed) and create new run context systematically based on those variables (or ExecutionContext).

**Comment by Ben8t** [comment]
@loicmathieu is still valid after recent changes?

**Comment by loicmathieu** [comment]
Yes, this is still something we want to tackle with @fhussonnois
